text/0066-separate-context-propagation.md

```diff
@@ -0,0 +1,606 @@
+between these events. For example, determining that a particular browser version 
+is associated with a failure in an image processing service.
+
+Correlations are implemented as an extremely basic cross-cutting concern - a 
+transaction-level dictionary. All key-value pairs are stored in the context 
+object, and then injected by the Correlation propagator to be sent to the next 
+process. 
+
+The Correlations API is based on the [W3C Correlation-Context specification](https://w3c.github.io/correlation-context/), 
+and implements the protocol as it is defined in that working group.
+
+While Correlations can be used to prototype other cross-cutting concerns, this 
+mechanism is primarily intended to convey values for the OpenTelemetry 
+observability systems. New concerns with different criteria should be modeled 
+separately, using the same underlying context propagation layer as building 
+blocks.
+
+For backwards compatibility, OpenTracing Baggage is propagated as Correlations 
+when using the OpenTracing bridge.
+
+As every correlation adds additional overhead to every request, correlations 
+should be used with care. The implementation MUST propagate these values. It is 
+entirely on the user to choose when to drop correlations, either by removing 
+them via the API, or by filtering them via a network proxy or service mesh.
+
+**`GetCorrelation(context, key) -> value`**  
+To access the value for a label set by a prior event, the Correlations API 
+provides a function which takes a context and a key as input, and returns a value.
+
+**`SetCorrelation(context, key, value) -> context`**  
+To record the value for a label, the Correlations API provides a function which 
+takes a context, a key, and a value as input, and returns an updated context 
+which contains the new value.
+
+**`RemoveCorrelation(context, key) -> context`**  
+To delete a label, the Correlations API provides a function 
+which takes a context and a key as input, and returns an updated context which 
+no longer contains the selected key-value pair.
+
+**`ClearCorrelations(context) -> context`**  
+To avoid sending any labels to an untrusted process, the Correlation API 
+provides a function to remove all Correlations from a context.
+
+**`GetCorrelationPropagator() -> (HTTP_Extractor, HTTP_Injector)`**  
+To deserialize the previous labels set by prior processes, and to serialize the 
+current total set of labels and send them to the next process, the Correlations 
+API provides a function which returns a Correlation-specific implementation of 
+the `HTTPExtract` and `HTTPInject` functions found in the Propagation API.
+
+# Context Propagation
+
+## Context API
+
+Cross-cutting concerns access data in-process using the same, shared context 
+object. Each concern uses its own namespaced set of keys in the context, 
+containing all of the data for that cross-cutting concern.
+
+**`GetValue(context, key) -> value`**  
```

Comment:
You need a GetKey (create Key api). Also a very important property that almost all the context implementation share is that the key is not identified by the string name, but by the instance returned by the GetKey to avoid issues when same string is used between different libraries.

Reply:
Ok I understand what you're talking about, adding it now.